streaming/stream_reason.hh

```diff
@@ -22,6 +22,7 @@ enum class stream_reason : uint8_t {
     rebuild,
     repair,
     replace,
+    tablet,
```

Comment:
That's not a reason. We're migrating a tablet, not tableting anything.

Reply:
Ok, will change to tablet_migration